src/main/java/org/assertj/core/api/ErrorCollector.java

```diff
@@ -36,24 +39,28 @@
   private final LastResult lastResult = new LastResult();
 
   @RuntimeType
-  public Object intercept(@This Object obj, @SuperCall Callable<Object> proxy,
-                          @SuperMethod(nullIfImpossible = true) Method method) throws Exception {
+  public static Object intercept(
+    @FieldValue(FIELD_NAME) ErrorCollector errorCollector,
+    @This Object obj,
+    @SuperCall Callable<?> proxy,
+    @SuperMethod(nullIfImpossible = true) Method method,
+    @StubValue Object stub) throws Exception {
     try {
       Object returnResult = proxy.call();
-      lastResult.setSuccess(true);
+      errorCollector.lastResult.setSuccess(true);
       return returnResult;
     } catch (AssertionError e) {
-      if (isNestedErrorCollectorProxyCall()) {
+      if (errorCollector.isNestedErrorCollectorProxyCall()) {
         // let the most outer call handle the assertion error
         throw e;
       }
-      lastResult.setSuccess(false);
-      errors.add(e);
+      errorCollector.lastResult.setSuccess(false);
+      errorCollector.errors.add(e);
     }
     if (method != null && !method.getReturnType().isInstance(obj)) {
       // In case the object is not an instance of the return type, just return null to avoid ClassCastException
       // with ByteBuddy
-      return null;
+      return stub;
```

Comment:
I'll remove the comment above which does not make sense

Reply:
This is to cover for primitive return values.